src/SimpleThings/EntityAudit/AuditReader.php

```diff
@@ -819,4 +819,39 @@ public function getEntityHistory($className, $id)
 
         return $result;
     }
+    
+    public function findRevisionsByDateAndUsername($className, $id, \DateTime $dateFrom = null, \DateTime $dateTo = null, $username = null)
```

Comment:
I've added a RevisionCriteria class like you said which contains the array_filter process. Is that what you meant in your comment?